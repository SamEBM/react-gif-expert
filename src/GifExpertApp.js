import React, { useState } from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    // const categories = ['Naruto', 'Demon Slayer', 'Dragon Ball', 'Attack on Titan'];
    const [categories, setCategories] = useState(defaultCategories);
    const handleAdd = (category) => {
        // No se deben mutar las variables
        // categories.push('Categoria Nueva');
        setCategories(prev => {
            return [category, ...prev];
        });
    };

    return (
        <>
            <h2>GifExpertApp</h2>
            <p>Ingresa el nombre de lo que quieres buscar:</p>
            <AddCategory addCategory={handleAdd} />
            <hr></hr>
            <ol>
                {categories.map( (category, index) => {
                    return <GifGrid key={category} category={category} />
                })}
            </ol>
        </>
    )
}

export default GifExpertApp