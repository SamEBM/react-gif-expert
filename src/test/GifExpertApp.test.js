import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas con la app completa', () => { 
    test('Debe mostrar <GifExpertApp /> correctamente', () => { 
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => { 
        const categories = ['Vegeta', 'Goku'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
})