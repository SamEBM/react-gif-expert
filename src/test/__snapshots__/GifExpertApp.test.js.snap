// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con la app completa Debe mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <p>
    Ingresa el nombre de lo que quieres buscar:
  </p>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas con la app completa Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <p>
    Ingresa el nombre de lo que quieres buscar:
  </p>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Vegeta"
      key="Vegeta"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
